### PR TITLE
Cleanup log file source invalidation

### PR DESCRIPTION
**Pull request by mnshdw**
### What does this PR do?

Cleaned up the code that avoid to evict invalid sources from the pool of sources in the file tailers.

### CODE REVIEW

File: pkg/logs/config/status.go [hunk 1]
```diff
@@ -10,61 +10,61 @@ import (
...
 	s.mu.Lock()
 	defer s.mu.Unlock()
-	s.pending = false
+	s.status = isError
 	s.error = fmt.Sprintf("Error: %s", err.Error())
 }
 
 // IsPending returns whether the current status is not yet determined.
 func (s *LogStatus) IsPending() bool {
-	s.mu.Lock()
```
Comment (mnshdw) [inline pkg/logs/config/status.go]: I don't sync we need to lock for read accesses

File: pkg/logs/input/file/file_provider.go [hunk 2]
```diff
@@ -55,7 +55,7 @@ func (p *Provider) FilesToTail() []*File {
 	shouldLogErrors := p.shouldLogErrors
 	p.shouldLogErrors = false // Let's log errors on first run only
 
-	sources := p.sources.GetValidSourcesWithType(config.FileType)
+	sources := p.sources.GetSourcesWithType(config.FileType)
```
Comment (ajacquemot) [inline pkg/logs/input/file/file_provider.go]: I think we don't need `GetValidSourcesWithType` anymore and replace all calls by `GetSourcesWithType `

**Review by ajacquemot — COMMENTED** [review]
Left two comments, otherwise looks good.

File: pkg/logs/config/status.go [hunk 3]
```diff
@@ -10,61 +10,61 @@ import (
 	"sync"
 )
 
+type status int
+
+const (
+	isPending status = iota
```
[the remark this answers is not in the document]
Comment (mnshdw) [reply pkg/logs/config/status.go]: `error` is a reserved keyword in go so I went for consistency

File: pkg/logs/input/file/file_provider.go (hunk 2 above)
Comment (mnshdw) [reply pkg/logs/input/file/file_provider.go]: Hmm we still want to filter out sources that stem from bad configurations in other places I think, I will check

**Pull request merged**